### PR TITLE
feat(feedback): add page for feedback architecture

### LINKED ISSUE #1208: User Feedback - Application documentation
Document the endpoints, infrastructure, consumers, etc.

Relates to:
* https://github.com/getsentry/develop/issues/1206

### PR DESCRIPTION
**Pull request by aliu39**
Exported from notion page in Engineering > Session Replay, with some improvements and omissions (rate limiting + deny list)
Page at /feedback-architecture

Todo:
- [x] embed mermaid diagrams
    - [x] make color scheme light mode
- [x] add to left sidebar
- [x] mention at end of User report section how Postgres model is used (issues > feedback tab)
- [x] list the 4 feature-related flags/options for self hosted users

closes https://github.com/getsentry/develop/issues/1208

### CODE REVIEW

**Comment by aliu39** [comment]
Screenshot 2024-06-18 at 4 35 19 PM
Need to figure out how to change the color scheme, used the /architecture page as a reference but the boxes are all dark..

File: src/docs/feedback-architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+release, url, etc.
+
+`USER_REPORT_ENVELOPE`: The older format with name/email/comments, that requires 
+`event_id` to link a Sentry error event.
+
+`USER_REPORT_DJANGO_ENDPOINT`: [The Web API](https://docs.sentry.io/api/projects/submit-user-feedback/)
+
+`CRASH_REPORT_EMBED_FORM`: The [crash report modal](https://docs.sentry.io/product/user-feedback/#crash-report-modal)
+
+## How it's stored
```
Comment (michellewzhang) [inline src/docs/feedback-architecture.mdx:28]:
```suggestion
## How feedback is stored
```

File: src/docs/feedback-architecture.mdx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+---
+
+## Feedback events
+
+The new and preferred way to send feedback from the SDK is as an [event item](https://develop.sentry.dev/sdk/envelopes/#full-examples), 
+in a MessagePack'ed envelope. The item type is the same as errors, but event 
+type = `"feedback"`. While user reports have an associated event, 
+**new feedback _is_ an event**. This offers 2 improvements:
+
+1. Users can submit generic feedback without an error occurring. The point of this feature is to catch things Sentry can’t!
```
Comment (michellewzhang) [inline src/docs/feedback-architecture.mdx:45]:
```suggestion
The new and preferred way to send feedback from the SDK is as an [event item](https://develop.sentry.dev/sdk/envelopes/#full-examples), 
in a MessagePack'ed envelope. The item type is the same as errors, except the event 
type = `"feedback"`. While user reports have an associated event, 
**new feedback _is_ an event**. This offers 2 improvements:

1. Users can submit generic feedback without an error occurring. The point of this feature is to allow users to catch things that Sentry can’t!
```

File: src/docs/feedback-architecture.mdx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+(`ingest-events`).
+<br/>
+
+### Attachments
+
+We only use attachments for the widget’s screenshot feature, which allows users 
+to submit **at most 1 screenshot per feedback**. Attachments are another [item type](https://develop.sentry.dev/sdk/envelopes/#attachment) 
+in an envelope.
+- SDK v8.0.0+, Relay v24.5.1+: Sends the feedback and attachment items in the same envelope.
+- SDK < v8, all Relay versions: Send a separate envelope for each item.
```
Comment (michellewzhang) [inline src/docs/feedback-architecture.mdx:84]:
```suggestion
- SDK v8.0.0+, Relay v24.5.1+: Sends the feedback and attachment items in the same envelope.
- SDK < v8, all Relay versions: Sends a separate envelope for each item.
```

File: src/docs/feedback-architecture.mdx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+### [`save_userreport()`](https://github.com/getsentry/sentry/blob/2b642e149c79b251e1c2f4339fc73d656347d74e/src/sentry/ingest/userreport.py#L28-L28)
+
+Before it was extended to generic feedback, the [`UserReport` model](https://github.com/getsentry/sentry/blob/2b642e149c79b251e1c2f4339fc73d656347d74e/src/sentry/models/userreport.py#L9-L9) 
+was first used for crash reports. Therefore an associated event id is required, and
```
Comment (michellewzhang) [inline src/docs/feedback-architecture.mdx:113]:
```suggestion
was first used for crash reports. Therefore an associated event ID is required, and 
```

File: src/docs/feedback-architecture.mdx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+The SDK function that sends these is `captureUserFeedback`.
+<br/>
+
+### Django endpoint
+
+Before our envelope ingest service existed, older SDKs directly POST the report 
+to Sentry, with
+`/api/0/projects/{organization_id_or_slug}/{project_id_or_slug}/user-feedback/`
+
+See [https://docs.sentry.io/api/projects/submit-user-feedback/](https://docs.sentry.io/api/projects/submit-user-feedback/)
```
Comment (michellewzhang) [inline src/docs/feedback-architecture.mdx:140]:
```suggestion
Before our envelope ingest service existed, older SDKs directly POST the report 
to Sentry, with
`/api/0/projects/{organization_id_or_slug}/{project_id_or_slug}/user-feedback/`.

See [https://docs.sentry.io/api/projects/submit-user-feedback/](https://docs.sentry.io/api/projects/submit-user-feedback/).
```

File: src/docs/feedback-architecture.mdx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+<br/>
+
+### Crash reports
+
+The crash report modal is a Django view -- an HTML form your app can render when 
+a page crashes. This was the earliest way of collecting feedback, and is for 
+error-specific feedback. You can install it as an [SDK integration](https://docs.sentry.io/platforms/javascript/user-feedback/#crash-report-modal).
```
Comment (michellewzhang) [inline src/docs/feedback-architecture.mdx:147]:
```suggestion
submitting feedback when Sentry detects an error. You can install it as an [SDK integration](https://docs.sentry.io/platforms/javascript/user-feedback/#crash-report-modal).
```

**Review by michellewzhang — COMMENTED** [review]
looks good so far! some small wording suggestions

File: src/docs/feedback-architecture.mdx [hunk 7]
```diff
@@ -0,0 +1,168 @@
+---
+title: User Feedback - Backend Architecture
+---
+
+**The goal of this doc is to give engineers an in-depth, systems-level understanding of User Feedback.**
+It will
```
Comment (c298lee) [inline src/docs/feedback-architecture.mdx:6]:
```suggestion
It will:
```

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**